### PR TITLE
1 config

### PR DESCRIPTION
**Pull request by imikejackson**
Dan, 
   We are using the Predicates code in our open-source product http://github.com/bluequartzsoftware/dream3d and need these changes to allow easier configuration of our build system to include Predicates. If you have any questions about the changes I have implemented just let me know. I need to test on MSVC which may need some minor changes to the codes in order to compile and export symbols as a DLL.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,62 @@
...
+    CACHE PATH
+    "Single Directory for all static libraries."
+    )
+endif()
+
+# Build shared libraries
+OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
+set(LIB_TYPE STATIC)
+if(BUILD_SHARED_LIBS)
+  #message(FATAL_ERROR "Due to issues exporting symbols with MSVC DREAM3D does NOT currently support intermediate shared libraries. Please turn OFF BUILD_SHARED_LIBS")
```
Comment (danshapero) [inline CMakeLists.txt]: This comment should be removed as it explicitly mentions DREAM3D.

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,4 +66,4 @@ enable_testing ()
 add_subdirectory (test)
 
 install (DIRECTORY ${PROJECT_SOURCE_DIR}/include/
-  DESTINATION include)
+         DESTINATION include/Predicates)
```
Comment (danshapero) [inline CMakeLists.txt]: Minor nitpick, but could you change this to lower-case "predicates" instead?

**Comment by danshapero** [comment]
Hi @imikejackson , thanks for the PR! I've been meaning to add a CMake config file for a while but haven't gotten around to it. I've commented on some of the changes, but once those are addressed it all looks good to me.

**Comment by imikejackson** [comment]
Changes have been pushed.

**Comment by danshapero** [comment]
I should also mention licensing concerns before you use this in your work. The copyright notice in the original code says only "Placed in the public domain by Jonathan Richard Shewchuk". [Some folks](http://www.rosenlaw.com/lj16.htm) are of the opinion that this is not a license at all, and that one should instead use a permissive license like BSD.

Seeing as his intention is clearly to let everyone use the work with no restrictions, I doubt that the original author is going to come after you with a pack of lawyers, but I thought you should be made aware so you can decide accordingyl. I emailed Dr. Shewchuk a year or so ago about this and didn't get a reply. If you have concerns about the licensing, I'd be happy to write to Dr. Shewchuk again and see if he'll instead release it under the BSD license or some appropriate equivalent.

**Comment by danshapero** [comment]
Great! Please rebase into a single commit and I'll merge.

**Comment by imikejackson** [comment]
Yes, I have attempted to contact Dr. Shewchuk in the past regarding some code and he never replied.

Are your intentions to place this code under a BSD license? If not then which license?

**Comment by imikejackson** [comment]
Ok. I rebased all 4 commits onto a single commit and pushed. Hope that was correct.

**Comment by danshapero** [comment]
Yes I was going to release it under BSD if I can.

[This project](https://github.com/otherlab/geode) has a reimplementation in C++ of exact geometric predicates as outlined in Shewchuk's papers from the 90s, and it's released under a BSD license. There's also [CoreLib](http://cs.nyu.edu/exact/core_pages/downloads.html), which is released under the QPL.

**Pull request merged**